ssz/tools/io.py

```diff
@@ -0,0 +1,150 @@
+from collections.abc import (
+    Mapping,
+    Sequence,
+)
+
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+)
+
+from ssz.sedes import (
+    BaseSedes,
+    Boolean,
+    ByteList,
+    ByteVector,
+    Container,
+    List,
+    UInt,
+    Vector,
+)
+from ssz.sedes.serializable import (
+    BaseSerializable,
+    MetaSerializable,
+)
+
+
+class FormattedDictIO:
+    def __init__(self, int_as):
+        self.int_as = int_as
+
+    def _parse_int(self, value):
+        if self.int_as == "int":
+            if not isinstance(value, int):
+                raise ValueError(f"Expected value of type int, got {type(value)}")
+            return value
+        elif self.int_as == "str":
+            if not isinstance(value, str):
+                raise ValueError(f"Expected value of type str, got {type(value)}")
+            return int(value)
+        elif self.int_as == "hex":
+            return int.from_bytes(decode_hex(value), "little")
+        else:
+            raise TypeError('Expected "int", "str", or "hex"')
+
+    def _dump_int(self, value, size):
+        if self.int_as == "int":
+            return value
+        elif self.int_as == "str":
+            return str(value)
+        elif self.int_as == "hex":
+            return encode_hex(value.to_bytes(size, "little"))
+        else:
+            raise TypeError('Expected "int", "str", or "hex"')
+
+    def parse_value(self, value, sedes=None):
```

Comment:
It doesn't appear this function correctly handles the case for `sedes == None` anywhere.